### PR TITLE
Remove Tools, add SonarQube

### PR DESCRIPTION
**Pull request by parc-jason**
Remove tools, reference showcase tools repo.  Add sonarqube to pipeline, add some simple render tests.

# Description
Use the standalone DevOps Tools repository and remove this repos tools files.  Add Jest Tests and SonarQube analysis to the pipeline.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
Unit tests are super simple, just ensure that our components will render without crashing.

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-52

File: Jenkinsfile.cicd [hunk 1]
```diff
@@ -83,6 +90,45 @@ pipeline {
...
+              timeout(10) {
+              echo 'Installing NPM Dependencies...'
+              sh 'npm ci'
+
+              echo 'Linting and Testing Frontend...'
+              sh 'npm run test'
+
+              echo 'Frontend Lint Checks and Tests passed'
+
+              }
```
Comment (jujaga) [inline Jenkinsfile.cicd]: There's missing indentation for lines 102-108 for the timeout contents, and an unnecessary whitespace line on 109.

File: frontend/src/auth/AuthService.jsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export default class AuthService {
   signinRedirectCallback = () => {
     this.UserManager.signinRedirectCallback()
       .then(() => {
-        '';
+        console.log('signin redirect callback executed');
```
Comment (jujaga) [inline frontend/src/auth/AuthService.jsx]: `// eslint-disable-line no-console`

File: Jenkinsfile [hunk 3]
```diff
@@ -306,7 +354,20 @@ pipeline {
...
+
+                  SonarQube: {
+                    unstash FE_COV_STASH
+
+                    echo 'Performing SonarQube static code analysis...'
+                    sh """
+                    sonar-scanner \
+                      -Dsonar.host.url='${SONARQUBE_URL_INT}' \
+                      -Dsonar.projectKey='${REPO_NAME}-${JOB_NAME}' \
+                      -Dsonar.projectName='NR Get Token (${JOB_NAME.toUpperCase()})'
```
Comment (loneil) [inline Jenkinsfile]: Sonar project name still refers to GETOK

File: Jenkinsfile.cicd [hunk 4]
```diff
@@ -301,7 +347,20 @@ pipeline {
...
+
+                  SonarQube: {
+                    unstash FE_COV_STASH
+
+                    echo 'Performing SonarQube static code analysis...'
+                    sh """
+                    sonar-scanner \
+                      -Dsonar.host.url='${SONARQUBE_URL_INT}' \
+                      -Dsonar.projectKey='${REPO_NAME}-${JOB_NAME}' \
+                      -Dsonar.projectName='NR Get Token (${JOB_NAME.toUpperCase()})'
```
Comment (loneil) [inline Jenkinsfile.cicd]: Still referring to GETOK

File: Jenkinsfile [hunk 5]
```diff
@@ -88,6 +97,45 @@ pipeline {
...
+              timeout(10) {
+              echo 'Installing NPM Dependencies...'
+              sh 'npm ci'
+
+              echo 'Linting and Testing Frontend...'
+              sh 'npm run test'
+
+              echo 'Frontend Lint Checks and Tests passed'
+
+              }
```
[the remark this answers is not in the document]
Comment (parc-jason) [reply Jenkinsfile]: pushing fix

File: Jenkinsfile.cicd (hunk 1 above)
Comment (parc-jason) [reply Jenkinsfile.cicd]: pushing fix

File: frontend/src/auth/AuthService.jsx (hunk 2 above)
Comment (parc-jason) [reply frontend/src/auth/AuthService.jsx]: complete file is /*eslint-disable */, lots of console.log...

File: Jenkinsfile.cicd (hunk 4 above)
Comment (parc-jason) [reply Jenkinsfile.cicd]: pushing in fix

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-52
